internal/pkg/deploy/cloudformation/stack/transformers.go

```diff
@@ -797,15 +797,26 @@ func convertPublish(topics []manifest.Topic, accountID, region, app, env, svc st
 	var publishers template.PublishOpts
 	// convert the topics to template Topics
 	for _, topic := range topics {
+		var fifoConfig *template.FIFOTopicConfig
+		if !topic.FIFO.IsEmpty() {
+			if topic.FIFO.IsEnabled() {
```

Comment:
I feel like the difference between `IsEmpty` and `IsEnabled` is a little subtle, what if the implementation of `IsEnabled` was like this:
```go
func (cfg FIFOAdvanceConfigOrBool) IsEnabled() bool {
   return aws.BoolValue(cfg.Enabled) || !cfg.Advanced.IsEmpty()
}
```
and then we can replace the `IsEmpty` check here with `IsEnabled`